tests/bcl-test/System.Security/System.Security-mac.csproj.template

```diff
@@ -0,0 +1,94 @@
+    <WarningLevel>4</WarningLevel>
+    <ConsolePause>false</ConsolePause>
+    <EnablePackageSigning>false</EnablePackageSigning>
+    <CodeSigningKey>Mac Developer</CodeSigningKey>
+    <EnableCodeSigning>false</EnableCodeSigning>
+    <CreatePackage>false</CreatePackage>
+    <LinkMode>None</LinkMode>
+    <XamMacArch>x86_64</XamMacArch>
+    <I18n>cjk,mideast,other,rare,west</I18n>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <DebugType>full</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release</OutputPath>
+    <DefineConstants>NET_1_1;NET_2_0;NET_3_0;NET_3_5;NET_4_0;NET_4_5;MONOMAC;DISABLE_CAS_USE;NO_GUI_TESTING;$(DefineConstants)</DefineConstants>
+    <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <ConsolePause>false</ConsolePause>
+    <LinkMode>None</LinkMode>
+    <CreatePackage>false</CreatePackage>
+    <CodeSigningKey>Mac Developer</CodeSigningKey>
+    <EnableCodeSigning>false</EnableCodeSigning>
+    <EnablePackageSigning>false</EnablePackageSigning>
+    <I18n>cjk,mideast,other,rare,west</I18n>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="System" />
+    <Reference Include="System.Xml" />
+    <Reference Include="System.Core" />
+    <Reference Include="System.Security" />
+    <Reference Include="Xamarin.Mac" />
+    <Reference Include="GuiUnit">
+      <HintPath>..\..\..\external\guiunit\src\framework\obj\Debug\net-4.5\GuiUnit.exe</HintPath>
+    </Reference>
+  </ItemGroup>
+  <ItemGroup>
+    <Folder Include="Resources\" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="Info-mac.plist">
+      <LogicalName>Info.plist</LogicalName>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="../../common/mac/MacTestMain.cs">
+      <Link>MacTestMain.cs</Link>
+    </Compile>
+    <Compile Include="..\..\common\Assert.cs">
+      <Link>Assert.cs</Link>
+    </Compile>
+  </ItemGroup>
+  <ItemGroup>
+#FILES#
+  </ItemGroup>
+  <ItemGroup>
+    <FilesToCopyEncryptedXml Include="..\..\..\external\mono\mcs\class\System.Security\Test\System.Security.Cryptography.Xml\sample.pfx" />
+    <FilesToCopyEncryptedXml Include="..\..\..\external\mono\mcs\class\System.Security\Test\System.Security.Cryptography.Xml\EncryptedXmlSample1.xml" />
+    <FilesToCopyEncryptedXml Include="..\..\..\external\mono\mcs\class\System.Security\Test\System.Security.Cryptography.Xml\EncryptedXmlSample2.xml" />
+    <FilesToCopyEncryptedXml Include="..\..\..\external\mono\mcs\class\System.Security\Test\System.Security.Cryptography.Xml\EncryptedXmlSample3.xml" />
+    <FilesToCopyCryptographyPkcs Include="..\..\..\external\mono\mcs\class\System.Security\Test\System.Security.Cryptography.Pkcs\detached.data" />
+    <FilesToCopyCryptographyPkcs Include="..\..\..\external\mono\mcs\class\System.Security\Test\System.Security.Cryptography.Pkcs\detached.p7" />
+  </ItemGroup>
```

Comment:
The above file group does not seem to be in the iOS project ?

Reply:
I assume it's only used for the larger profile exposed by XM ?
That makes sense because XM Full has 492 vs XI 125 tests.

However XM Modern also has 492 tests ? and that should be nearly identical to mobile ? @chamons can you check that

Reply:
In general each project flavor should have the file list matching it, and Modern should match iOS in almost all cases.

However, the way I handle generating two projects (Full and Modern) assumes we start from the full project and strip it down to Modern. This means secondary included files like this XML unnecessarily come for the ride.

In mono here:

https://github.com/mono/mono/tree/master/mcs/class/System.Security

no body defines a source list for tests except the default System.Security_test.dll.sources.

This makes me suspect any difference in tests is based on # defines. @mandel-macaque can diff the test list to track it down.

Reply:
@spouliot @chamons let me explain. System.Security in iOS is a subset, that is, we do not support all classes. Those classes that we do not support are not present in the dll and therefore the tests are ignored with a !MOBILE (from a mono PR I did a while back). The item group you see there is adding files that are required for the tests that are ignore in iOS, that is, System.Security.Cryptography.Pkcs and System.Security.Cryptography.Xml are not present in the mobile version and therefore the tests do not need those tests files. Modern does have the tests because the classes are present, so unless I understand modern wrong, this is correct and expected.